### PR TITLE
Verification preview

### LINKED ISSUE #4764: Natively Produce versions of tables 2-8 of the CI Verification Report
We should provide a method of displaying the content of (at least) tables 2 thru 8 of the verification report from within the CI tools of Ilios. The tables would be available to view during report development as well as post-submission.

Table 1 May not be high value, except maybe to alert of mapping gaps.
Tables 9-10 are duplicative of our existing `program year objective` mapping report. We may want to simply expand/align that tool.

**Table 1: Program Expectations Mapped to PCRS**

> The Program Expectations ID is generated by AAMC. It is referenced again in Table 9, 10 and 12. A single Program Expectation can be related to multiple PCRS, and a single
> PCRS can be related to multiple Program Expectations.

**Table 2: Primary Instructional Method by Non-Clerkship Sequence Block**

> Table 2 may be used to support your response to the LCME DCI Tables 6.0-1 and 6.0-2.
> The CI Standardized Vocabulary contains many instructional method choices. Instructional methods are grouped here to (mimic, and) assist in completing the LCME DCI. 

**Table 3-A: Non-Clerkship Sequence Block Instructional Time**

> The amount of time (Total Weeks) in a Sequence Block is calculated using duration (the number of days divided into 5-day weeks). For example, a 75-day duration Sequence Block will be calculated as 15 weeks. Duration is an optional field for non-clerkship sequence blocks. Only Sequence Blocks with duration values provided will appear in table 3-A.
> The Average Hours of Instruction Per Week is calculated by summing the total event duration (hours, minutes) for events tagged with an instructional method divided by the number of weeks in the Sequence Block (as calculated above). This table displays events that are tagged with instructional methods. It does not include events that are tagged with only assessment methods (i.e., Assessment Events).

**Table 3-B: Clerkship Sequence Block Instructional Time**

> Table 3-B may be used to support your response to the LCME DCI Table 6.0-3.
> The amount of time (Total Weeks) in a Clerkship Sequence Block is calculated using duration (the number of days divided into 5-day weeks). For example, a 75-day duration Clerkship Sequence Block will be calculated as 15 weeks. Duration is a required field for clerkship sequence blocks and is provided by you.
> The Average Hours of Instruction Per Week is calculated by summing the total event duration (hours, minutes) for events tagged with an instructional method divided by the number of weeks in the Clerkship Sequence Block. This table displays events that are tagged with instructional methods. It does not include events that are tagged with only assessment methods (i.e., Assessment Events).

**Table 4: Instructional Method Counts**

> This table shows the number of times each Instructional Method was used; this differs from Table 2 which groups some instructional methods. Events used in multiple sequence
> blocks will only have their instructional methods counted once in this table so the sum reflects total number of instructional events in your curriculum. 

N.B.: Ilios does not provide for Non-Primary Method assignment, so that value will always be zero. We should include "Academic Level" as a column value instead.

**Table 5: Non-Clerkship Sequence Block Assessment Methods**

> Table 5 may be used to support your response to the LCME DCI Tables 9.0-1 and 9.0-2.
> Every Sequence Block that is not a clerkship and contains at least one Assessment Event will appear in this table. An Assessment Event may contain more than one assessment method.;
> Sequence Blocks that do not have Assessment Events will not appear.
> Select assessment methods are grouped to assist in completing the LCME DCI. Number of exams is calculated by totaling the number Assessment Events in a sequence block that have assessment methods tagged as summative. When a specific assessment method is employed in a Sequence Block, the corresponding grouping will get an X. If at least one assessment method in an event in the Sequence Block is tagged as formative, the Formative Assessment (Y/N) column will get a Y. If at least one event in the Sequence Block contains AM010, Narrative Assessment Narrative Assessment (Y/N) will get a Y.

**Table 6: Clerkship Sequence Block Assessment Methods**

> Table 6 may be used to support your response to the LCME DCI Table 9.0-3.
> Every Sequence Block that is a clerkship and contains one Assessment Event will appear in this table; Sequence Blocks that do not have Assessment Events will not appear.
> Select assessment methods are grouped to assist in completing the LCME DCI. Number of exams is calculated by totaling the number Assessment Events in a sequence block that have assessment methods tagged as summative.. When a specific assessment method is employed in a Sequence Block, the corresponding grouping will get an X. If at least one assessment method in an event in the Sequence Block is tagged as formative, the Formative Assessment (Y/N) column will get a Y. If at least one event in the Sequence Block contains AM010, Narrative Assessment, the Narrative Assessment (Y/N) will get a Y.

**Table 7: All Events with Assessments Tagged as Formative or Summative**

> This table counts all assessment methods tagged within events. A single event used in multiple sequence blocks will only be counted once in this table. 
> The Number of Summative Assessments, when summed, represents the total number of assessment methods tagged as summative in your curriculum; the Number of Formative Assessments, when summed, represents the total number of assessment methods tagged as formative in your curriculum.

**Table 8: All Resource Types**

> All resources tagged in each event are counted, including multiple occurrences of the same resource within a single event. 

N.B.: We should include "Academic Level" as a column value.
[Verification Report-Tables_1-9.pdf](https://github.com/ilios/frontend/files/3498422/Verification.Report-Tables_1-9.pdf)

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes #4764

### CODE REVIEW

File: app/templates/components/curriculum-inventory-verification-preview.hbs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    <li><a href="#verification-preview-table3b">{{t "general.table3bClerkshipSequenceBlockInstructionalTime"}}</a></li>
+    <li><a href="#verification-preview-table4">{{t "general.table4InstructionalMethodCounts"}}</a></li>
+    <li>
+      <a href="#verification-preview-table5">{{t "general.table5NonClerkshipSequenceBlockAssessmentMethods"}}</a>
+    </li>
+    <li><a href="#verification-preview-table6">{{t "general.table6ClerkshipSequenceBlockAssessmentMethods"}}</a></li>
+    <li>
+      <a href="#verification-preview-table7">
+        {{t "general.table7AllEventsWithAssessmentsTaggedAsFormativeOrSummative"}}
+      </a><
```
Comment (jrjohnson) [inline app/templates/components/curriculum-inventory-verification-preview.hbs]: Seems like some cruft here?

File: tests/integration/components/curriculum-inventory-report-overview-test.js [hunk 2]
```diff
@@ -67,6 +67,7 @@ module('Integration | Component | curriculum inventory report overview', functio
     await render(hbs`{{curriculum-inventory-report-overview report=report canUpdate=true}}`);
     return settled().then(() => {
       assert.dom('.title').hasText('Overview', 'Component title is visible.');
+      assert.dom('.report-overview-actions .verification-preview').exists({ count: 1 }, 'Rollover course button is visible.');
```
Comment (jrjohnson) [inline tests/integration/components/curriculum-inventory-report-overview-test.js]: Copy paste `'Rollover course button is visible.'` test name.

File: app/templates/components/curriculum-inventory-verification-preview-table1.hbs [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+              <li>{{pcrs}}</li>
+            {{/each}}
+          </ul>
+        </td>
+      </tr>
+    {{/each}}
+  </tbody>
+</table>
+
+{{#if @tocId}}
```
Comment (jrjohnson) [inline app/templates/components/curriculum-inventory-verification-preview-table1.hbs]: I'm not sure I understand this, why not a fixed ID and why an `if` is it ever not present?

File: tests/integration/components/curriculum-inventory-report-overview-test.js (hunk 2 above)
Comment (stopfstedt) [reply tests/integration/components/curriculum-inventory-report-overview-test.js]: fixed.

File: app/templates/components/curriculum-inventory-verification-preview.hbs (hunk 1 above)
Comment (stopfstedt) [reply app/templates/components/curriculum-inventory-verification-preview.hbs]: nice catch. fixed.

File: app/templates/components/curriculum-inventory-verification-preview-table1.hbs (hunk 3 above)
Comment (stopfstedt) [reply app/templates/components/curriculum-inventory-verification-preview-table1.hbs]: i suppose i could simplify this. the whole linking business makes me a bit uneasy, as it creates these inter-dependencies between these components that fall outside the usual component nesting strategy.

Comment (stopfstedt) [reply app/templates/components/curriculum-inventory-verification-preview-table1.hbs]: refactored, i pulled those links up and out.

**Pull request merged**